src/transformers/modeling_utils.py

```diff
@@ -1437,6 +1437,16 @@ def __init__(self, config: PretrainedConfig, *inputs, **kwargs):
                 config, torch_dtype=torch.get_default_dtype(), check_device_map=False
             )
         self.config = config
+        if getattr(config, "loss_type", None) is None:
+            loss_type = self.__class__.__name__
+            if loss_type not in LOSS_MAPPING:
+                loss_groups = f"({'|'.join(LOSS_MAPPING)})"
+                loss_type = re.findall(loss_groups, self.__class__.__name__)
+                if len(loss_type) > 0:
+                    loss_type = loss_type[0]
+                else:
+                    loss_type = None
+            self.config.loss_type = loss_type
```

Comment:
As 
```
            loss_type = self.__class__.__name__
            if loss_type not in LOSS_MAPPING:
                loss_groups = f"({'|'.join(LOSS_MAPPING)})"
                loss_type = re.findall(loss_groups, self.__class__.__name__)
                if len(loss_type) > 0:
                    loss_type = loss_type[0]
                else:
                    loss_type = None
```

is used twice we can probably have a function for `find_class_loss_function`and keep a single source of truth! 🤗 

https://github.com/huggingface/transformers/blob/00962d103459218c6f353da319a1d12adcd56a15/src/transformers/modeling_utils.py#L5044-L5051

Reply:
I was thinking that if we are going to set the loss_type into the config at init time, we can remove the entire check from the loss_function method and simplify it to the following:
```
    def loss_function(self):
        loss_type = getattr(self.config, "loss_type", None)

        if loss_type is None or loss_type not in LOSS_MAPPING:
            logger.warning_once(
                f"`loss_type={loss_type}` was set in the config but it is unrecognised."
                f"Using the default loss: `ForCausalLMLoss`."
            )
            loss_type = "ForCausalLM"
        return LOSS_MAPPING[loss_type]
```
What do you think?

Reply:
yep sounds good!

Reply:
Done!